pkg/tasks/update_template_distributions.go

```diff
@@ -329,3 +347,215 @@ func getRHRepoContentPath(rawURL string) (string, error) {
 	}
 	return u.Path[1 : len(u.Path)-1], nil
 }
+
+// RunCandlepin creates an environment for the template and content sets for each repository.
+// Each content set's URL is the distribution URL created during RunPulp().
+// May promote or demote content from the environment, depending on if the repository is being added or removed from template.
+// If not created for the given org previously, this will also create a product and pool.
+func (t *UpdateTemplateDistributions) RunCandlepin() error {
```

Comment:
I saw two ways to combine the tasks here: 
1. Interweave the functionality and modify pulp/candlepin in the same method run. This might reduce the number of for loops by a little. But for candlepin we make use of bulk APIs, so the only place that would really feel the benefit of this is creating content sets.
2. Do everything in pulp, then do everything in candlepin. This way separates the candlepin logic from the pulp logic, which I feel makes the flow much easier to follow.